### PR TITLE
Feature/characters gameplay

### PR DESCRIPTION
**Pull request by JorxSS**
What's new:

- Singleplayer & Multiplayer logic implemented
- Combat
- AI Battle circle
- FSM

### CODE REVIEW

File: Engine/Component/ComponentCollider.cpp [hunk 1]
```diff
@@ -373,20 +377,32 @@ void ComponentCollider::SetRollingFriction()
...
+{
+	return center;
+}
+
 void ComponentCollider::SetVelocity(float3& velocity, float speed)
 {
 	//bottom of the model
 	btVector3 bottom = body->getWorldTransform().getOrigin();
-	bottom.setY(bottom.getY() - 2*box_size.getY());
+	bottom.setY(bottom.getY() - (5 * box_size.getY()));
```
Comment (mariofv) [inline Engine/Component/ComponentCollider.cpp:405]: What does this 5 mean?

File: Engine/Component/ComponentCollider.cpp [hunk 2]
```diff
@@ -418,6 +434,76 @@ void ComponentCollider::SetVelocity(float3& velocity, float speed)
 
 }
 
+void ComponentCollider::SetVelocityEnemy(float3 & velocity, float speed)
+{
+	//bottom of the model
+	btVector3 bottom = body->getWorldTransform().getOrigin();
+	bottom.setY(bottom.getY() - 200 * box_size.getY());
```
Comment (mariofv) [inline Engine/Component/ComponentCollider.cpp:441]: What does this number mean?

File: Game/Assets/Scripts/src/Script/CameraController.cpp [hunk 3]
```diff
@@ -95,22 +95,23 @@ void CameraController::OnInspector(ImGuiContext* context)
...
+	ImGui::Checkbox("Freeze", &freeze);
 }
 
 void CameraController::ActivePlayer()
 {
 	if (god_mode) 
 	{
 		player_movement_component->Disable();
-		if(App->input->singleplayer_input)
+		if(!App->input->singleplayer_input)
```
Comment (mariofv) [inline Game/Assets/Scripts/src/Script/CameraController.cpp:106]: This is kinda strange, we should have a look into this before VS3

File: Game/Assets/Scripts/src/Script/PursueEnemyState.cpp [hunk 4]
```diff
@@ -32,11 +32,29 @@ void PursueEnemyState::OnStateEnter()
...
+	//{
+	//	if (!enemy->animation->IsOnState("Idle"))
+	//	{
+	//		enemy->animation->ActiveAnimation("idle");
+	//	}
+
+	//	Exit(enemy->idle_state);
+	//}
+	enemy->BattleCircleAI();
+
```
Comment (mariofv) [inline Game/Assets/Scripts/src/Script/PursueEnemyState.cpp:56]: If this is not used anymore it should be deleted

File: Game/Assets/Scripts/src/Script/WorldManager.h [hunk 5]
```diff
@@ -5,6 +5,8 @@
 #include "EventManager.h"
 
 class ComponentCollider;
+class PlayerController;
+class ComponentImage;
```
Comment (mariofv) [inline Game/Assets/Scripts/src/Script/WorldManager.h:9]: Code Style ®️

File: cpp.hint [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// Hint files help the Visual Studio IDE interpret Visual C++ identifiers
+// such as names of functions and macros.
+// For more information see https://go.microsoft.com/fwlink/?linkid=865984
+#define ENGINE_API __declspec(dllexport)
+#define ENGINE_API __declspec(dllimport)
```
Comment (mariofv) [inline cpp.hint:5]: What is this? 🤔

**Review by mariofv — APPROVED** [review]
Looks good!

File: Engine/Component/ComponentCollider.cpp (hunk 2 above)
Comment (vidalmelero) [reply Engine/Component/ComponentCollider.cpp:441]: It is used to make bigger the ray in the raycast of bullet to detect the ground. At 5 should be enough.

**Comment by vidalmelero** [comment]
Good job!

**Review by vidalmelero — APPROVED** [review]

**Pull request merged**